akka-remote/src/main/scala/akka/remote/artery/SystemMessageDelivery.scala

```diff
@@ -42,7 +42,19 @@ private[remote] object SystemMessageDelivery {
   final case class Ack(seqNo: Long, from: UniqueAddress) extends Reply
   final case class Nack(seqNo: Long, from: UniqueAddress) extends Reply
 
-  final case object ClearSystemMessageDelivery
+  /**
+   * Sent when an incarnation of an Association is quarantined. Consumed by the
+   * SystemMessageDelivery stage on the sending side, i.e. not sent to remote system.
+   * The SystemMessageDelivery stage will clear the sequence number and other state associated
+   * with that incarnation.
+   *
+   * The incarnation counter is bumped when the handshake is completed, so a new incarnation
+   * corresponds to a new UID of the remote system.
+   *
+   * The SystemMessageDelivery stage also detects that the incarnation has changed when sending or resending
+   * system messages.
+   */
+  final case class ClearSystemMessageDelivery(incarnation: Int)
```

Comment:
Thanks for adding docs on this one, can be a tricky one to figure out otherwise :)